### PR TITLE
(feat) O3-4194 Ward App - add ability to cancel pending admit / trans…

### PR DESCRIPTION
**Pull request by chibongho**
…fer requests

## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [x] My work includes tests or is validated by existing tests.

## Summary

This PR adds the ability for us to cancel pending admit or transfer requests of a patient.

## Screenshots

image
image
image

## Related Issue

https://openmrs.atlassian.net/browse/O3-4194

### CODE REVIEW

File: packages/esm-ward-app/src/ward-workspace/cancel-admission-request-workspace/cancel-admission-request.workspace.tsx [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+          concept: emrConfiguration?.admissionDecisionConcept?.uuid,
+          value: {
+            uuid: emrConfiguration?.denyAdmissionConcept?.uuid,
+          },
+        },
+      ];
+
+      createEncounter(patient, emrConfiguration?.cancelADTRequestEncounterType, [
+        {
+          concept: emrConfiguration?.dispositionDescriptor?.dispositionSetConcept?.uuid,
```
Comment (mogoodrich) [inline packages/esm-ward-app/src/ward-workspace/cancel-admission-request-workspace/cancel-admission-request.workspace.tsx:82]: This actually doesn't belong with an disposition set. The obs can just be stored directly on on the encounter.

**Review by mogoodrich — CHANGES_REQUESTED** [review]
Generally looks good @chibongho  but see my comment about the structure of the deny admission obs... it's just a stand-alone question and answer, not a disposition.

File: packages/esm-ward-app/src/ward-workspace/cancel-admission-request-workspace/cancel-admission-request.test.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
+import { useAppContext, type DefaultWorkspaceProps } from '@openmrs/esm-framework';
+import { screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+import { mockInpatientRequestAlice, mockLocationInpatientWard, mockPatientAlice } from '../../../../../__mocks__';
+import { renderWithSwr } from '../../../../../tools';
```
Comment (denniskigen) [inline packages/esm-ward-app/src/ward-workspace/cancel-admission-request-workspace/cancel-admission-request.test.tsx:6]:
```suggestion
import { mockInpatientRequestAlice, mockLocationInpatientWard, mockPatientAlice } from '__mocks__';
import { renderWithSwr } from 'tools';
```

File: packages/esm-ward-app/src/ward-patient-card/row-elements/ward-patient-pending-transfer.tsx [hunk 3]
```diff
@@ -25,12 +28,27 @@ const WardPatientPendingTransfer: React.FC<WardPatientTransferProps> = ({ wardPa
...
+  };
+
   if (!(dispositionType === 'TRANSFER' || dispositionType === 'DISCHARGE')) return null;
 
   return (
     <div className={styles.wardPatientCardDispositionTypeContainer}>
-      <Movement className={styles.movementIcon} size="24" />
+      <Movement className={styles.movementIcon} size={24} />
       {message}
+      <Button
```
Comment (denniskigen) [inline packages/esm-ward-app/src/ward-patient-card/row-elements/ward-patient-pending-transfer.tsx:44]: nit: It's preferable to use [IconButton](https://react.carbondesignsystem.com/?path=/docs/components-combobox--overview) for the icon-only button variant.

**Review by denniskigen — COMMENTED** [review]
Just a couple of thoughts

File: packages/esm-ward-app/src/ward-workspace/cancel-admission-request-workspace/cancel-admission-request.workspace.tsx (hunk 1 above)
Comment (chibongho) [reply packages/esm-ward-app/src/ward-workspace/cancel-admission-request-workspace/cancel-admission-request.workspace.tsx:82]: Wait... I think that's what I'm doing? This array is an array of obs for the encounter.

Comment (chibongho) [reply packages/esm-ward-app/src/ward-workspace/cancel-admission-request-workspace/cancel-admission-request.workspace.tsx:82]: nvm... I think I know what you mean now. It should be just `obs` instead.

File: packages/esm-ward-app/src/ward-workspace/cancel-admission-request-workspace/cancel-admission-request.test.tsx
Comment (denniskigen) [inline packages/esm-ward-app/src/ward-workspace/cancel-admission-request-workspace/cancel-admission-request.test.tsx:1]: 👨🏼‍🍳

**Review by denniskigen — COMMENTED** [review]
LGTM, but I defer to @mogoodrich on the deny admission obs stuff.

File: packages/esm-ward-app/src/ward-workspace/cancel-admission-request-workspace/cancel-admission-request.workspace.tsx [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+        },
+        {
+          concept: emrConfiguration?.admissionDecisionConcept?.uuid,
+          value: {
+            uuid: emrConfiguration?.denyAdmissionConcept?.uuid,
+          },
+        },
+      ];
+
+      createEncounter(patient, emrConfiguration?.cancelADTRequestEncounterType, obs)
```
Comment (mogoodrich) [inline packages/esm-ward-app/src/ward-workspace/cancel-admission-request-workspace/cancel-admission-request.workspace.tsx:80]: Yep, exactly!

**Review by mogoodrich — APPROVED** [review]
I didn't re-review the whole thing, but the change I commented on now looks correct, thanks!

**Pull request merged**